mpsl/include/mpsl_clock.h

```diff
@@ -0,0 +1,131 @@
+                                            intervals in any case.
+
+                                 @note Must be 0 if source is not ::MPSL_CLOCK_LF_SRC_RC.
+
+                                 @note The application must ensure calibration at least once every
+                                       8 seconds to ensure +/-500 ppm clock stability.
+                                       The recommended configuration for ::MPSL_CLOCK_LF_SRC_RC
+                                       is given by @ref MPSL_RECOMMENDED_RC_CTIV
+                                       and @ref MPSL_RECOMMENDED_RC_TEMP_CTIV.
+                                       This sets the calibration interval to 4 seconds and guarantees
+                                       clock calibration every second calibration interval. That is,
+                                       the clock will be calibrated every 8 seconds.
+                                       If the temperature changes more that 0.5 every 4 seconds, the
+                                       clock will be calibrated every 4 seconds. See the
+                                       Product Specification for more information. */
+} mpsl_clock_lf_cfg_t;
+
+
+/** @brief High frequency clock callback.
+ *
+ * This callback will be called when the high frequency clock is started.
+ * It will be executed in the same execution priority as @ref mpsl_low_priority_process.
+ */
+typedef void (*mpsl_hf_clock_callback_t)(void);
+
+/** @brief Request the high frequency crystal oscillator.
+ *
+ * Will start the high frequency crystal oscillator, the startup time of the crystal varies
+ * and the ::mpsl_clock_hfclk_is_running function can be polled to check if it has started.
+ *
+ * @see mpsl_clock_hfclk_is_running
+ * @see mpsl_clock_hfclk_release
+ *
+ * @param[in] hf_clock_started_callback Function to be called when the high frequency clock is started.
+ *                                      The callback will be executed in the context as
+ *                                      @ref mpsl_low_priority_process.
+ * @retval 0  Success
+ */
+int32_t mpsl_clock_hfclk_request(mpsl_hf_clock_callback_t hf_clock_started_callback);
+
+/** @brief Releases the high frequency crystal oscillator.
+ *
+ * Indicates that the high frequency crystal oscillator is not needed by the application.
+ * MPSL may continue to use the high frequency clock if it is requested by protocol stacks.
+ * MPSL will automatically turn it off when it is no longer needed.
+ *
+ * @see mpsl_clock_hfclk_is_running
+ * @see mpsl_clock_hfclk_request
+ *
+ * @retval 0  Success
+ */
+int32_t mpsl_clock_hfclk_release(void);
+
+/** @brief Checks if the high frequency crystal oscillator is running.
+ *
+ * @see mpsl_clock_hfclk_request
+ * @see mpsl_clock_hfclk_release
+ *
+ * @param[out] p_is_running 1 if the external crystal oscillator is running, 0 if not.
+ *
+ * @retval 0  Success
+ */
+int32_t mpsl_clock_hfclk_is_running(uint32_t * p_is_running);
```

Comment:
HFCLK is always running when the CPU is running, this function is testing whether HFINT (internal) or HFXO (crystal) is in use.

Should we instead rename to

mpsl_clock_hfclk_hfxo_is_running

?

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fps_nrf52840%2Fclock.html

It is confusing as the function is named now.

Reply:
Yes, I'll make an issue for it, so that we can maybe fix it for a later release.

Reply:
It looks strange to me, but if it is already convention to refer to HFXO as HFCLK then this is OK.